### PR TITLE
Initial observed state display

### PR DESCRIPTION
**Pull request by Jim3Things**
... and further detail info

Note that this is a first drop of a sequence, done so that we can make progress over the next week.

### CODE REVIEW

File: pkg/protos/inventory/external.proto [hunk 1]
```diff
@@ -15,19 +15,59 @@ option go_package = "github.com/Jim3Things/CloudChamber/simulation/pkg/protos/in
...
+
+        // Observed, actual, and target data follows on from here...
+    }
+
+    // Individual blade within the rack
+    message Blade {
+        BladeDetails details = 1;
+
+        BladeCapacity capacity = 2;
+
```
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply pkg/protos/inventory/external.proto:49]: Oops. Mis-spot. Sorry.

File: pkg/protos/inventory/common.ts_ref [hunk 2]
```diff
@@ -774,3 +919,10 @@ export const RootDetails = {
     return obj;
   },
 };
+
+// If you get a compile-error about 'Constructor<Long> and ... have no overlap',
+// add '--ts_proto_opt=esModuleInterop=true' as a flag when calling 'protoc'.
+if (util.Long !== Long) {
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/common.ts_ref:925]: We can when we redo makefile, but as we just use it for change testing it shouldn't matter.

File: pkg/protos/inventory/external.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/external.proto:49]: protoc, not go, type is first...

File: pkg/protos/inventory/external.tsx [hunk 3]
```diff
@@ -1,43 +1,128 @@
...
-        this.blades.set(Number(key), new BladeCapacity(value))
-      });
-    }
+    this.tors = new Map<number, External_Tor>()
+    this.fullBlades = new Map<number, External_Blade>()
+
+    this.blades = asMap(object.blades, (k, v) => [Number(k), new BladeCapacity(v)])
+    this.pdus = asMap(object.pdus, (k, v) => [Number(k), new External_Pdu(v)])
+    this.tors = asMap(object.tors, (k, v) => [Number(k), new External_Tor(v)])
+    this.fullBlades = asMap(object.fullBlades, (k, v) => [Number(k), new External_Blade(v)])
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/external.tsx:125]: yes it does, I'll delete those two lines.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**